substrate/frame/staking/src/pallet/impls.rs

```diff
@@ -832,7 +833,9 @@ impl<T: Config> Pallet<T> {
 		let mut nominators_taken = 0u32;
 		let mut min_active_stake = u64::MAX;
 
-		let mut sorted_voters = T::VoterList::iter();
+		// voter list also contains chilled/idle voters, filter those.
+		let mut sorted_voters =
+			T::VoterList::iter().filter(|v| Self::status(&v) != Ok(StakerStatus::Idle));
```

Comment:
Might be more explicit to filter like this?
```suggestion
			T::VoterList::iter().filter(|v| Self::status(&v) == Ok(StakerStatus::Nominator));
```